src/Interactive/Features/Interactive/Core/InteractiveHost.cs

```diff
@@ -248,50 +270,33 @@ private void DisposeChannel()
                 _serverChannel = null;
             }
         }
-
-        public TextWriter Output
+        public void SetOutput(TextWriter value)
         {
-            get
+            if (value == null)
             {
-                return _output;
+                throw new ArgumentNullException(nameof(value));
             }
 
-            set
-            {
-                if (value == null)
-                {
-                    throw new ArgumentNullException(nameof(value));
-                }
-
-                var oldOutput = Interlocked.Exchange(ref _output, value);
-                oldOutput.Flush();
-            }
+            var oldOutput = Interlocked.Exchange(ref _output, value);
+            oldOutput.Flush();
         }
 
-        public TextWriter ErrorOutput
+        public void SetErrorOutput(TextWriter value)
         {
-            get
+            if (value == null)
             {
-                return _errorOutput;
+                throw new ArgumentNullException(nameof(value));
             }
 
-            set
-            {
-                if (value == null)
-                {
-                    throw new ArgumentNullException(nameof(value));
-                }
-
-                var oldOutput = Interlocked.Exchange(ref _errorOutput, value);
-                oldOutput.Flush();
-            }
+            var oldOutput = Interlocked.Exchange(ref _errorOutput, value);
+            oldOutput.Flush();
         }
 
         internal void OnOutputReceived(bool error, char[] buffer, int count)
         {
             (error ? ErrorOutputReceived : OutputReceived)?.Invoke(buffer, count);
 
-            var writer = error ? ErrorOutput : Output;
+            var writer = error ? _errorOutput : _output;
             writer.Write(buffer, 0, count);
         }
```

Comment:
Do we need to lock around `writer.Write()`?

Reply:
Yes.

---
In reply to: [188322596](https://github.com/dotnet/roslyn/pull/26770#discussion_r188322596) [](ancestors = 188322596)